crate/operator/data/watermark_collector.yaml

```diff
@@ -0,0 +1,30 @@
+---
```

Comment:
I think the filename needs to be `watermark-collector.yaml` (`-` instead of `_`)